tests/shared/sdk/usbboot.spec.js

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const m = require('mochainon')
+const _ = require('lodash')
+const Bluebird = require('bluebird')
+const usb = require('../../../lib/shared/sdk/utils/usb')
+const usbboot = require('../../../lib/shared/sdk/usbboot')
+
+const createUSBDeviceStub = (options) => {
+  return {
+    deviceDescriptor: {
+      idVendor: options.vendorID,
+      idProduct: options.productID,
+      iManufacturer: options.manufacturerIndex,
+      iProduct: options.productIndex
+    },
+    _configDescriptor: {
+      bNumInterfaces: options.interfaces
+    },
+    open: _.noop,
+    close: _.noop,
+    getStringDescriptor: (index, callback) => {
+      callback(null, options.strings[index])
+    }
+  }
+}
+
+const BCM2835_STUB = createUSBDeviceStub({
+  vendorID: 0x0a5c,
+  productID: 0x2763,
+  manufacturerIndex: 1,
+  productIndex: 2,
+  interfaces: 1,
+  strings: [
+    null,
+    'Broadcom',
+    'BCM2708 Boot'
+  ]
+})
+
+const BCM2837_STUB = createUSBDeviceStub({
```

Comment:
According to the logfiles I gathered, the CM3 (BCM2837) always starts up with `bNumInterfaces: 1`, but according to a [code comment](https://github.com/raspberrypi/usbboot/blob/master/main.c#L108) in the upstream `rpiboot` there _might_ also be some circumstances where the BCM2837 starts up with two interfaces. Perhaps it'd be worth also creating and testing a `BCM2837_TWO_INTERFACES_STUB` ?